### PR TITLE
add heat equation example

### PR DESCRIPTION
**Pull request by jw3126**
Here is the heat equation example talked about in #73. It is somewhat inaccurate:

image

About 2.5% error. Also error does not get better if I use a finer grid.

@ChrisRackauckas you mentioned that passing a mass matrix might help, can you elaborate?

Code to create the plot:
```julia
@time include("examples/heat_equation.jl")
import PyPlot
import FEniCS
import .HeatEquationExample
Ex = HeatEquationExample
display(FEniCS.Plot(Ex.p.u))
display(FEniCS.Plot(Ex.u_true))
```

### CODE REVIEW

File: examples/heat_equation.jl [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+end
+
+u0_vec = get_array(u)
+t_final = 0.1
+tspan = (0.0, t_final)
+prob = ODEProblem(f!, u0_vec, tspan, p)
+u_true = interpolate(Expression(u_str, degree=2, pi=Float64(pi), t=t_final), V)
+
+alg = ImplicitEuler(autodiff=false)
+sol = OrdinaryDiffEq.solve(prob, alg)
```
Comment (ChrisRackauckas) [inline examples/heat_equation.jl]:
```suggestion
sol = OrdinaryDiffEq.solve(prob, alg, reltol=1e-6)
```

? It might just be a tolerance thing.

Comment (ChrisRackauckas) [reply examples/heat_equation.jl]: Note that `ImplicitEuler` is adaptive

Comment (jw3126) [reply examples/heat_equation.jl]: If I change `reltol=1e-6` and refine the grid to 50 I get an error of 0.2%. 
If I change `reltol=1e-6` and refine the grid to 1000 I get an error of 0.7%.
Is it expected that refining the grid can increase error in this setup?

Comment (ChrisRackauckas) [reply examples/heat_equation.jl]: not really, but with that tolerance there's a random of errors you'd expect. Drop it even lower then and it should go down more: `reltol=1e-9,abstol=1e-9`. ImplicitEuler is really slow though, so you might want to use `Rosenbrock23(autodiff=false)`, `Rodas5(autodiff=false)`, or `KenCarp4(autodiff=false)` when you get to that.

Comment (jw3126) [reply examples/heat_equation.jl]: Ok thanks! It's possible to get the error small this way. I go with 1% error to keep the example fast.

**Comment by jw3126** [comment]
@ChrisRackauckas does the code look sane to you now?

**Pull request merged**